Make unit test run with `catkin_make run_tests`

sloretz opened the pull request:
I notice the unit tests wasn't running when using `catkin_make run_tests`. I changed how the test was added to use the catkin cmake macros. I also added python-mock as a test dependency because the official ubuntu/trusty docker image does not have that installed by default.

sloretz committed:
Make unit test run with `catkin_make run_tests`

sloretz committed:
Remove find_program(python)

Catkin is already required, and requires python

clalancette commented:
I love the change.  However, it looks like all CI has failed, with no yaml.  I think we'll need to add python-yaml to the package.xml as well.

sloretz committed:
Removed plain CMake install steps in favor of catkin's install steps

Moved catkin_python_setup() after catkin_package()

sloretz committed:
added python-yaml as a run-depend

clalancette commented:
A couple of things:

- While you are messing around with the package.xml, it might be worthwhile to convert it over to format 2, to make it easier to integrate with ROS2.
- Travis is still failing, not being able to find urdf_parser_py.  I've seen that locally when I forget to source devel/setup.bash in my catkin workspace; I think something similar must be going on with travis.  Whether it is worthwhile to fix travis, however, I'm not sure.

sloretz committed:
Added src/ to sys.path

The unit tests can fail in some cases with catkin_make.
When using catkin_make the unit tests cannot depend on an environment
being set up with a complete environment (thankes @dirk-thomas!)
This was not an issue with catkin_make_isolated or `catkin build`

sloretz commented:
@clalancette Will do with the package.xml!

I learned from @dirk-thomas that unit tests can't expect to have the python path set up for importing python packages from the current package in all cases when using `catkin_make`. The unit test now adds `../src` to sys.path. `catkin_make_isolated` and `catkin build` don't have this issue.

sloretz committed:
Updated package.xml to format 2

sloretz commented:
@clalancette I updated the package.xml. An odd thing is it has a `<depend>python</depend>`. I'm 99% sure that is unnecessary.

clalancette commented:
I agree with you that the <depend> on python is a bit weird.  However, there are other packages (vision_opencv, for instance), which also have that dependency.  I'll leave it up to you, but I'd just leave it in for now; it probably doesn't hurt anything.  Otherwise, I think this looks good to me, so feel free to merge it.
